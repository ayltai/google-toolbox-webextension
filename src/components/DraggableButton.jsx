import { Avatar, GridListTile, IconButton, makeStyles, Tooltip, } from '@material-ui/core';
import PropTypes from 'prop-types';
import React from 'react';
import { SortableElement, } from 'react-sortable-hoc';

export const DraggableButton = SortableElement(({ service, configuration, }) => {
    const useStyles = makeStyles(theme => ({
        iconButton : {
            margin : theme.spacing(1),
        },
        avatar     : {
            width  : configuration.size,
            height : configuration.size,
        },
    }));

    const classes = useStyles();

    return (
        <GridListTile>
            <Tooltip title={service.name}>
                <IconButton
                    className={classes.iconButton}
                    size='small'>
                    <Avatar
                        className={classes.avatar}
                        variant='square'
                        src={`images/${service.id}.png`}>
                        {service.name.substr(0, 1)}
                    </Avatar>
                </IconButton>
            </Tooltip>
        </GridListTile>
    );
});

DraggableButton.propTypes = {
    index         : PropTypes.number.isRequired,
    service       : PropTypes.object.isRequired,
    configuration : PropTypes.object.isRequired,
};
