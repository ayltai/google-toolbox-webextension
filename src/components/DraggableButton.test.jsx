import { mount } from 'enzyme';
import React from 'react';
import { SortableContainer } from 'react-sortable-hoc';
import renderer from 'react-test-renderer';

import { DraggableButton } from './DraggableButton';

const service = {
    id   : 'android-auto',
    name : 'Android Auto',
    url  : 'https://www.android.com/intl/en_us/auto/',
};

const configuration = {
    size : 48,
    cols : 4,
    rows : 2,
};

describe('<DraggableButton />', () => {
    it('renders correctly', () => {
        expect(renderer.create(
            <SortableContainer>
                <DraggableButton
                    index={0}
                    service={service}
                    configuration={configuration} />
            </SortableContainer>
        ).toJSON()).toMatchSnapshot();
    });

    it('mounts without error', () => {
        mount(
            <SortableContainer>
                <DraggableButton
                    index={0}
                    service={service}
                    configuration={configuration} />
            </SortableContainer>
        );
    });
});
