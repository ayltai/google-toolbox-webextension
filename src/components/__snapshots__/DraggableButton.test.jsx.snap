// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DraggableButton /> renders correctly 1`] = `null`;
